applications/FluidDynamicsHydraulicsApplication/custom_utilities/hydraulic_fluid_auxiliary_utilities.cpp

```diff
@@ -307,4 +307,60 @@ void HydraulicFluidAuxiliaryUtilities::SetInletFreeSurface(ModelPart &rModelPart
     });
 }
 
+void HydraulicFluidAuxiliaryUtilities::CalculateArtificialViscosity(ModelPart &rModelPart,double WaterDynamicViscosityMax){
+    const auto &r_process_info = rModelPart.GetProcessInfo();
+
+    block_for_each(rModelPart.Elements(), [&](Element &rElement)
+    {
+    double artificial_viscosity;
+
+    rElement.Calculate(ARTIFICIAL_DYNAMIC_VISCOSITY,artificial_viscosity ,r_process_info);
+        if (artificial_viscosity > WaterDynamicViscosityMax)
+        {
+            artificial_viscosity = WaterDynamicViscosityMax;
+        }
+        double neg_nodes = 0.0;
+        double pos_nodes=0.0;
+        for (auto &r_node : rElement.GetGeometry())
+        {
+            double distance = r_node.FastGetSolutionStepValue(DISTANCE);
+            
+            if (distance > 0)
+            {
+                pos_nodes += 1;
+            }
+            else
+            {
+                neg_nodes += 1;
+            }
+        }
+        if (neg_nodes > 0 && pos_nodes > 0)
+        {
+            artificial_viscosity = 0.0;
+        }
```

Comment:
I'd first check if it is whether it is intersected or not. If it is not, you calculate it. If not you skip it.

Reply:
Not for this PR but we should at some point do an `IsCut` tiny function in the utilities (we're doing this in many places).